nixos/modules/services/networking/hostapd.nix

```diff
@@ -1,222 +1,1283 @@
-          Warning: This passphrase will get put into a world-readable file in
-          the Nix store!
-        '';
-      };
+                IEEE 802.11d: This advertises the countryCode and the set of allowed channels
+                and transmit power levels based on the regulatory limits.
 
-      logLevel = mkOption {
-        default = 2;
-        type = types.int;
-        description = lib.mdDoc ''
-          Levels (minimum value for logged events):
-          0 = verbose debugging
-          1 = debugging
-          2 = informational messages
-          3 = notification
-          4 = warning
-        '';
-      };
+                IEEE802.11h: This enables radar detection and DFS (Dynamic Frequency Selection)
+                support if available. DFS support is required on outdoor 5 GHz channels in most
+                countries of the world.
+              '';
+            };
 
-      countryCode = mkOption {
-        default = null;
-        example = "US";
-        type = with types; nullOr str;
-        description = lib.mdDoc ''
-          Country code (ISO/IEC 3166-1). Used to set regulatory domain.
-          Set as needed to indicate country in which device is operating.
-          This can limit available channels and transmit power.
-          These two octets are used as the first two octets of the Country String
-          (dot11CountryString).
-          If set this enables IEEE 802.11d. This advertises the countryCode and
-          the set of allowed channels and transmit power levels based on the
-          regulatory limits.
-        '';
-      };
+            hwMode = mkOption {
+              default = "g";
+              type = types.enum ["a" "b" "g" "ad" "any"];
+              description = mdDoc ''
+                Operation mode (a = IEEE 802.11a (5 GHz), b = IEEE 802.11b (2.4 GHz),
+                g = IEEE 802.11g (2.4 GHz), ad = IEEE 802.11ad (60 GHz); a/g options are used
+                with IEEE 802.11n (HT), too, to specify band). For IEEE 802.11ac (VHT), this
+                needs to be set to hw_mode=a. For IEEE 802.11ax (HE) on 6 GHz this needs
+                to be set to hw_mode=a. When using ACS (see `channel` parameter), the
+                special value "any" can be used to indicate that any supported band may be selected.
+                This special case is currently only supported with hardware for which the driver
+                implements offloaded ACS.
+
+                Most likely you want to select 'a' (5GHz & 6GHz a/n/ac/ax) or 'g' (2.4GHz b/g/n) here.
+              '';
+            };
+
+            channel = mkOption {
+              default = 7;
+              example = 11;
+              type = types.int;
+              description = mdDoc ''
+                The channel to operate on. Use 0 to enable ACS (Automatic Channel Selection).
+                Beware that not every device supports ACS in which case {command}`hostapd`
```

Comment:
How common is that? If the devices are in the wast minority which do not support this we might want to set it to 0 by default

Reply:
Very uncommon.

Reply:
Some devices support it, but the vast majority of kernel drivers does *not* implement it. You want to use it when you can, but you'll likely only be able to use it on very well supported chipsets. `mt7921` based cards works for example.